CMakeLists.txt

```diff
@@ -141,6 +141,26 @@ if(ccache)
   endif()
 endif()
 
+#---Enable distcc ------------------------------------------------------------------------------
+if(distcc)
+  find_program(DISTCC_COMMAND NAMES distcc)
+  mark_as_advanced(DISTCC_COMMAND ${DISTCC_COMMAND})
+  if(EXISTS ${DISTCC_COMMAND})
+    message(STATUS "Found distcc: ${DISTCC_COMMAND}")
+    if (ccache)
+        # If ccache is enabled, use distcc as CCACHE_PREFIX
+        set_property(GLOBAL PROPERTY RULE_LAUNCH_COMPILE "CCACHE_PREFIX=${DISTCC_COMMAND} ${CCACHE_COMMAND}")
```

Comment:
Can we do this safely?

Reply:
Looks good for me! Maybe @amadio has more ideas how to improve code..
(Can we add some message for ccache  that it is using distcc? I think it will be not clear for user if he managed to setup everything properly..)

Reply:
I never used distcc with ccache, so I don't know. Depending on the shell, the variable assignment may not work. Try with `fish` and `tcsh`. If those work, then it should be fine. Otherwise, you may have to add an `env ...` to the beginning of the command line.

Reply:
Using the `CCACHE_PREFIX` hook for `distcc` is fine, I think. But as @amadio said, I don't understand whether cmake is in this stage depended on the underlying shell.

Reply:
I'll try with tcsh and fish, if this works I would also say it's gonna be fine.

Reply:
Ok, works also with fish and tcsh :) Also added a message if distcc is used in combination with ccache and the `CCACHE_PREFIX` is used. @oshadura Good to go?